src/Features/Core/Portable/AddParameter/AbstractAddParameterCodeFixProvider.cs

```diff
@@ -69,15 +72,55 @@ internal abstract class AbstractAddParameterCodeFixProvider<
                 return null;
             }
 
+            if (this.CannotConvertDiagnosticIds.Contains(diagnostic.Id))
+            {
+                return null;
+            }
+
             return initialNode.GetAncestorsOrThis<TArgumentSyntax>()
                               .LastOrDefault(a => a.AncestorsAndSelf().Contains(node));
         }
 
-        private Task HandleInvocationExpressionAsync(
+        private async Task HandleInvocationExpressionAsync(
             CodeFixContext context, TInvocationExpressionSyntax invocationExpression, TArgumentSyntax argumentOpt)
         {
-            // Currently we only support this for 'new obj' calls.
-            return SpecializedTasks.EmptyTask;
+            var document = context.Document;
+            var cancellationToken = context.CancellationToken;
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+
+            var expression = syntaxFacts.GetExpressionOfInvocationExpression(invocationExpression);
+            if (expression == null)
+            {
+                return;
+            }
+
+            var symbolInfo = semanticModel.GetSymbolInfo(expression, cancellationToken);
+            var candidates = symbolInfo.CandidateSymbols.OfType<IMethodSymbol>().ToImmutableArray();
+
+            if (candidates.Length == 0)
+            {
+                // Invocation might be on an delegate. We try to find the declaration of the delegate.
+                // This only fixes the declaration but not the delegate type. This ode is for test purposes only and will likely be removed.
```

Comment:
is this code needed anymore?

Reply:
No. I just kept it to illustrate that delegate invocations are going to be much more complicated and proposed in one of the comments to not taken those into account for this PR.